src/main/java/randoop/generation/ForwardGenerator.java

```diff
@@ -235,6 +236,17 @@ private void initializeRuntimePrimitivesSeen() {
 
     eSeq.execute(executionVisitor, checkGenerator);
 
+    // Dynamic type casting permits calling methods that do not exist on the declared type.
+    if (GenInputsAbstract.cast_to_run_time_type && eSeq.isNormalExecution()) {
+      Sequence oldSeq = eSeq.sequence;
+      castToRunTimeType(eSeq);
```

Comment:
Since this method may change its argument, I think it would be more appropriate as an instance method rather than a method defined outside `ExecutableSequence`.

Whether or not you do that, can some or all of the body of the `if` be folded into the routine, to make the call site cleaner and simpler?